### DISCUSSION TITLE
Auth: add_permission problem ...

### DISCUSSION THREAD

**Discussion by 1st-github-account** [emmett-framework/emmett#452] [Q&A] [answered]
Hi!
I want to use the authentization system of emmett (python 3.8 + emmett 2.4.12).
Creating users and groups/membership work fine. But I cannot add a permission.

```
@app.command('setup')
def setup():
    with db.connection():
        user = User.create(email="lu@localhost", first_name="Lu",last_name="",password="pw")
        admins = auth.create_group('admins')
        auth.add_membership(admins, user.id)
        auth.add_permission(admins, 'ban_users') # <========
        db.commit()
```

`$ emmett setup`
.... Error message: "AttributeError: 'Auth' object has no attribute 'add_permission'".

It seems that the function does not exist. What am I doing wrong?
Thanks a lot!
Lu

**Comment by gi0baro** [comment, marked as answer]
Hi @1st-github-account, this is actually wrong documented. Gonna update the docs ASAP.

You can convert your code to:

```python
@app.command('setup')
def setup():
    with db.connection():
        user = User.create(email="lu@localhost", first_name="Lu",last_name="",password="pw")
        admins = auth.create_group('admins')
        admins.users.add(user)
        admins.auth_permissions.create(name="ban_users")
        db.commit()
```

**Reply by 1st-github-account** [reply]
Hi @gi0baro, this works great!
Thanks a lot & have a nice day,
Lu
